.github/scripts/parse-issue-body.sh

```diff
@@ -12,7 +12,7 @@ body=${body//$'\r'/}
 body=${body//$'\n'/}
 body=${body//$'\r\n'/}
 
-regex="### GitHub Username[\r|\n|\r\n]*([0-9a-zA-Z-]*)[\r|\n|\r\n]*### Discord ID[\r|\n|\r\n]*(.{3,32}#[0-9]{4})"
+regex="### GitHub Username([0-9a-zA-Z-]*)### Discord ID(.{3,32}#[0-9]{4}|[^#@:\`]{3,32})"
 if [[ ! $body =~ $regex ]]; then
```

Comment:
I am confused as to how this can still match this input:

```
### GitHub Username
yihau
### Discord ID
chido6488
```

The regex does not allow for newlines.
`([0-9a-zA-Z-]*)` between the `### GitHub Username` and `### Discord ID` should not match `\nyihau\n`.
What am I missing? %)

It actually does not match for me:  http://tpcg.io/_VZOG8V
Also, the old version was using character sets incorrectly.
`|` is not a delimiter in a character set.
It should be just `[\r\n]*`, rather than `[\r|\n|\r\n]*`.
Or it should be `(\r|\n|\r\n)*`.
And, inside double-quoted strings, backslashes are processed before they reach the regex engine.  So I am puzzled as to how the old version worked.
If they both pass the tests, there must be something wrong with the tests.

I would suggest a regex to be written like this:
```bash
regex='### GitHub Username[[:space:]\r\n]*([0-9a-zA-Z-]*)[[:space:]\r\n]*### Discord ID[[:space:]\r\n]*(.{3,32}#[0-9]{4}|[^#@:`]{3,32})[[:space:]\r\n]*'
```

You can verify that this one works in the same bash playground: http://tpcg.io/_VZOG8V

Reply:
The regex can also be split into multiple lines for readability:

```bash
regex='### GitHub Username[[:space:]\r\n]*'
regex="$regex"'[0-9a-zA-Z-]*[[:space:]\r\n]*'
regex="$regex"'### Discord ID[[:space:]\r\n]*'
regex="$regex"'(.{3,32}#[0-9]{4}|[^#@:`]{3,32})[[:space:]\r\n]*'
```

Reply:
> The regex does not allow for newlines.
([0-9a-zA-Z-]*) between the ### GitHub Username and ### Discord ID should not match \nyihau\n.
What am I missing? %)


I think my regex works cuz I do this one in the previous lines
https://github.com/solana-labs/contributor-access-policy/blob/8e26dbf63cde5c162ea20ed291eb679815b01bcd/.github/scripts/parse-issue-body.sh#L11-L13

so actually the input will become

```
### GitHub Usernameyihau### Discord IDchido6488
```

Reply:
btw, you're right. I should remove line 13 I think. thank you for pointing this out <3

Reply:
Missed removal of the newlines, as I was assuming the previous version was actually using those `\n|\r` character sets at first %)

Sorry for the noise :)
But, consider using single quotes for regex expressions.
Would not have to escape backslashes, as they are quite common in the regex syntax.

Reply:
thank you for reviewing my PR. tbh, I love all your suggestions. will make some changes later. 🔨